andela/mervels-bn-backend
# #168131416 Set up authentication with Google and Facebook

#### What does this PR do?
Implement login with Facebook and Google
#### Description of Task to be completed?
- Write the endpoint that requests data from Google
- Write the endpoint that requests data from Facebook
- Write the controller that creates user or logs in user on successful social login
#### How should this be manually tested?
- Clone this repo
- Checkout to branch `ft-implement-social-login-168131416`
- Add environment variables
- Install dependencies i.e. `npm install`
- Run `npm test`
- Run the application with `npm run start:dev` and navigate to `api/v1/auth/google` or `api/v1/auth/facebook` in your browser to authenticate with Facebook or Google
- As an alternative, you can the Heroku review app to test this: `http://mervels-bn-backend-stagi-pr-15.herokuapp.com/api/v1/auth/google` and `http://mervels-bn-backend-stagi-pr-15.herokuapp.com/api/v1/auth/facebook`
- After you log in to your account you should see a response that looks like the one in screenshots
#### Any background context you want to provide?
- To have the client IDs and secret keys, you have to create apps in developer space either for google or facebook.
- Both on Facebook and Google you will have to add authorized redirect URLs i.e `http://localhost:3000/api/v1/auth/facebook/redirect` and `http://localhost:3000/api/v1/auth/google/redirect` respectively.
- Also, you will need to add `http://localhost:3000` to the authorized domains to allow it to send requests. You have to do this for both platforms.
- For Facebook, you need to add test users because it only allows localhost when using created test users for development.
#### What are the relevant pivotal tracker stories?
[#168131416](https://www.pivotaltracker.com/story/show/168131416)
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/50402526/64708354-c83aa880-d4b4-11e9-8b7e-0ac3af9e0eb1.png)

#### Questions:
N/A
